elastic/logstash/pipeline/logstash.conf

```diff
@@ -15,126 +17,197 @@ filter {
-        remove_field => "message"
-      }
       mutate {
-        add_field => { "read_timestamp" => "%{@timestamp}" }
-      }
-      date {
-        match => [ "[nginx][access][time]", "dd/MMM/YYYY:H:m:s Z" ]
-        remove_field => "[nginx][access][time]"
+        split => { "[@metadata][jwt]" => "." }
       }
-      useragent {
-        source => "[nginx][access][agent]"
-        target => "[nginx][access][user_agent]"
-        remove_field => "[nginx][access][agent]"
+
+      mutate {
+        copy => { "[@metadata][jwt][1]" => "[@metadata][b64]" }
       }
-      geoip {
-        source => "[nginx][access][remote_ip]"
-        target => "[nginx][access][geoip]"
+
+      ruby {
+        init => "require 'base64'"
+        code => "event.set('[@metadata][jwt_decoded]', Base64.decode64(event.get('[@metadata][b64]')))"
       }
 
-      geoip {
-        source => "[nginx][access][client_ip]"
-        target => "[nginx][access][server][geoip]"
+      json {
+        source => "[@metadata][jwt_decoded]"
+        target => "jwt_data"
       }
+    }
 
+    }
 
-      # decode JWT
-      if [@metadata][jwt] {
+    if [nginx][access][payload] {
+      # remove \ from payload
+      mutate {
+        gsub => [
+          "[nginx][access][payload]", "[\\]", ""
+        ]
+      }
 
+      # parse json to data field
+      json {
+        source => "[nginx][access][payload]"
+        target => "[data]"
+      }
+    }
 
-        mutate {
-          split => { "[@metadata][jwt]" => "." }
-        }
 
-        mutate {
-          copy => { "[@metadata][jwt][1]" => "[@metadata][b64]" }
-        }
+    # Get the type of operation of interest
 
-        ruby {
-          init => "require 'base64'"
-          code => "event.set('[@metadata][jwt_decoded]', Base64.decode64(event.get('[@metadata][b64]')))"
-        }
+    mutate {
+      add_field => { "[@metadata][action]" => "%{[nginx][access][method]} %{[nginx][access][url]}" }
+    }
 
-        json {
-          source => "[@metadata][jwt_decoded]"
-          target => "jwt_data"
-        }
+    # Get the operation
+    translate {
+      field => "[@metadata][action]"
+      destination => "operation"
+      dictionary_path => "/app/logstash/dictionaries/action.yml"
+      fallback => "other"
+      regex => true
+    }
 
+    # Extract mine guid based on operation
+    if [operation] != "other" {
+      grok {
+        match => { "[nginx][access][url]" => [".*/api/mines/%{UUID:[mine][id]}", ".*/api/documents/mines/%{UUID:[mine][id]}"] }
       }
 
-      if [nginx][access][payload] {
-        # remove \ from payload
-        mutate {
-          gsub => [
-            "[nginx][access][payload]", "[\\]", ""
-          ]
+      if [mine][id] {
+            # lookup by mine id
+        jdbc_streaming {
```

Comment:
Looking at the options for this plugin,  the default timeout for cache seems to be 5 seconds.  
https://www.elastic.co/guide/en/logstash/current/plugins-filters-jdbc_streaming.html#plugins-filters-jdbc_streaming-cache_expiration

I don't have . any strong feelings for what value this should be but maybe a minute or so? or longer?  mines don't change that much.
I don't know how many database hits we are talking so I may be wrong, @itchison any strong feelings?

Reply:
I have changed the default timeout to 300 seconds/ 5 minutes. Thank you.